core/control-plane/control-plane-aggregate-services/src/main/java/org/eclipse/edc/connector/service/contractnegotiation/ContractNegotiationProtocolServiceImpl.java

```diff
@@ -194,6 +204,17 @@ private ServiceResult<ContractNegotiation> validateRequest(ClaimToken claimToken
             return ServiceResult.success(negotiation);
         }
     }
+    
+    @NotNull
+    private ServiceResult<ContractNegotiation> validateGetRequest(ClaimToken claimToken, String id, ContractNegotiation negotiation) {
+        var result = validationService.validateRequest(claimToken, negotiation);
+        if (result.failed()) {
+            // should return not found if counter-party not authorized
```

Comment:
I know that this is stated in the specs, but shouldn't be more http-like to return 403?

Reply:
I think the idea behind returning the 404 is to not give out information about the existence of processes to unauthorized parties.